test/executor_test.rb

```diff
@@ -138,6 +138,7 @@ def plan
             wait_for('execution plan removed from executor') do
               !director.current_execution_plan_ids.include?(execution_plan.id)
             end
+            world.persistence.find_execution_plans(filters: { uuid: [execution_plan.id] }).must_be :empty?
```

Comment:
I was running into the issue that the procedure on failure was still trying to update the state of the execution plan (that it still had in memory), so it saved it with invalid state, and was causing issues in other parts: this way we make sure the execution plan doesn't re-appear.